cpp/src/arrow/dataset/test_util.h

```diff
@@ -60,34 +60,31 @@ class DatasetFixtureMixin : public ::testing::Test {
  public:
   DatasetFixtureMixin() : ctx_(std::make_shared<ScanContext>()) {}
 
- protected:
-  std::shared_ptr<ScanOptions> options_;
-  std::shared_ptr<ScanContext> ctx_;
-};
+  /// \brief Ensure that record batches found in reader are equals to the
+  /// record batches yielded by the data fragment.
+  void AssertScanTaskEquals(RecordBatchReader* expected, ScanTask* task) {
```

Comment:
At the end, this should also check that `expected` is exhausted (otherwise, the sequences were not the same length).